Mathlib/Lean/ToExpr.lean

```diff
@@ -0,0 +1,57 @@
+/-
+Copyright (c) 2025 Paul Lezeau. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Paul Lezeau, Bhavik Mehta
+-/
+import Mathlib.Data.Finset.Empty
+
+/-! # Helper functions to create Expr terms for Finsets, Multisets and Sets.
+
+This file provides some basic tools to create an `Expr` term representing a
+finset/multiset/set of terms of type `α`, given a `ToExpr` instance on `α`.
+
+-/
+
+universe u
+
+open Lean Meta
+
+/-- A helper function for building `Expr` terms from a list of terms of type `α`. -/
+private def List.toExprAux {α} [ToExpr α] (nilFn : Expr)
+    (singFn : Expr) (consFn : Expr) : List α → Expr
+  | []    => nilFn
+  | [a] => .app singFn (toExpr a)
+  | a::as => mkApp2 consFn (toExpr a) (List.toExprAux nilFn singFn consFn as)
+
+/-- Produces the `Expr` corresponding to the finset associated to a list.
+
+`List.toFinsetExpr l` is the `Expr` representing `l.toFinset`. -/
```

Comment:
```suggestion
`List.toFinsetExpr l` is the `Expr` representing `l.toFinset`. -/
```
this isn't quite accurate, that `Expr` would be `mkAppM `List.toFinset [toExpr l]` or something...
Instead, the point is that if `l` is an explicit list of things, this produces the explicit finset with the same elements

Reply:
Good point, fixed!